### PR TITLE
Update CI to Go 1.19 to resolve linter build failures.

### PR DESCRIPTION
**Pull request by matthewdale**
## Summary
Update the Evergreen CI to use Go 1.19 to run static analysis and tests.

## Background & Motivation
Motivated by build failures when trying to install the latest version of `golangci-lint`. Build failure:
```
go: downloading github.com/golangci/golangci-lint v1.51.1
...
# honnef.co/go/tools/unused
../../../pkg/mod/honnef.co/go/tools@v0.4.0/unused/unused.go:419:14: obj.Origin undefined (type *types.Var has no field or method Origin)
../../../pkg/mod/honnef.co/go/tools@v0.4.0/unused/unused.go:421:14: obj.Origin undefined (type *types.Func has no field or method Origin)
note: module requires Go 1.19
# honnef.co/go/tools/staticcheck
../../../pkg/mod/honnef.co/go/tools@v0.4.0/staticcheck/lint.go:4032:29: righti.Method(i).Origin undefined (type *types.Func has no field or method Origin)
../../../pkg/mod/honnef.co/go/tools@v0.4.0/staticcheck/lint.go:4037:36: sel.Obj().(*types.Func).Origin undefined (type *types.Func has no field or method Origin)
note: module requires Go 1.19
```

### CODE REVIEW

File: .evergreen/config.yml [hunk 1]
```diff
@@ -2121,19 +2121,19 @@ axes:
           - windows-64-vs2017-test
         variables:
           GCC_PATH: "/cygdrive/c/ProgramData/chocolatey/lib/mingw/tools/install/mingw64/bin"
-          GO_DIST: "C:\\golang\\go1.18"
+          GO_DIST: "C:\\golang\\go1.19"
           VENV_BIN_DIR: "Scripts"
       - id: "ubuntu1604-64-go-1-18"
```
Comment (prestonvasquez) [inline .evergreen/config.yml:2126]: The id values should updated, correct? i.e. "ubuntu1604-64-go-1-19"

I think this also requires updating the matrices, [here](https://github.com/mongodb/mongo-go-driver/blob/c9b3eb092798b2016dc3a7b8083ef3f707320bd9/.evergreen/config.yml#L2412).

**Review by prestonvasquez — CHANGES_REQUESTED** [review]

File: .evergreen/config.yml (hunk 1 above)
Comment (prestonvasquez) [reply .evergreen/config.yml:2126]: Then documentation [here](https://github.com/mongodb/mongo-go-driver/blame/master/README.md#L26) should also be updated.

**Review by kevinAlbs — APPROVED** [review]

File: .evergreen/config.yml (hunk 1 above)
Comment (prestonvasquez) [reply .evergreen/config.yml:2126]: I think the labels are entirely descriptive. I don't have a strong opinion, I suppose the argument for maintaining the Go Version is that it illustrates which version of Go we're testing on without having to visit the evergreen config file or read testing logs.

Comment (matthewdale) [reply .evergreen/config.yml:2126]: Sounds good, I'll keep it and update it.

**Review by prestonvasquez — APPROVED** [review]

**Pull request merged**